public/js/widget-manager.js

```diff
@@ -132,12 +154,53 @@ define(['jupyter-js-widgets'], function(Widgets) {
      * widgetNode: DOM node where the widget should render
      * outputAreaModel: OutputArea that contains the widget
      */
-    WidgetManager.prototype.trackPending = function(msg_id, widgetNode, outputAreaModel) {
+    WidgetManager.prototype.trackPending = function(kernelFuture, widgetNode, outputAreaModel) {
+        this._hookupDeclWidgetsCallbacks(kernelFuture, widgetNode, outputAreaModel);
+
+        var msg_id = kernelFuture.msg.header.msg_id;
         this._pendingExecutions[msg_id] = {
             widgetNode: widgetNode,
             outputAreaModel: outputAreaModel
         };
     };
 
+
+    /**
+     * DECLARATIVE WIDGETS SHIMS
+     **/
+
+    WidgetManager.prototype._shimDeclWidgets = function(kernel) {
+        var ipy = window.IPython = window.IPython || {};
+        var nb = ipy.notebook = ipy.notebook || {};
+        nb.events = nb.events || $({});
+
+        nb.kernel = kernel;
+        nb.kernel.is_connected = function() {
+            return kernel.status === KernelStatus.Busy || kernel.status === KernelStatus.Idle;
+        };
+        nb.kernel.widget_manager = this;
+
+        // IPython.notebook.base_url   ?????
+
+        // WidgetManager is instantiated after creation of a kernel, so assume it is ready
+        nb.events.trigger('kernel_ready.Kernel');
+    };
+
+    WidgetManager.prototype._hookupDeclWidgetsCallbacks = function(kernelFuture, widgetNode, outputAreaModel) {
+        var that = this;
+
+        kernelFuture.onReply = function(msg) {
+            window.IPython.notebook.events.trigger('shell_reply.Kernel');
+        };
+
+        // Declarative Widgets attempts to get `callbacks` through "cell" data
+        var $cell = $(widgetNode).parents('.cell');
+        var cellData = $cell.data('cell') || {};
+        cellData.get_callbacks = function() {
+            return that._get_callbacks(outputAreaModel);
+        };
+        $cell.data('cell', cellData);
```

Comment:
This is a mimic of what the notebook provides today. Not sure decl widgets can do anything differently at the moment if it is to remain compatible with existing notebook.